### PR TITLE
Properly position atomic inline level boxes

### LINKED ISSUE #25709: Text is not wrapped around div with replaced content in layout_2020
In the following test case the text appears on top of the div, but it should really be positioned under it.

```
<div><img style="background: pink; width: 100px; height: 100px;"></div>
<p>this should be under</p>
```

Output:
Screenshot from 2020-02-07 16-32-07

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #25709

### CODE REVIEW

File: components/layout_2020/flow/inline.rs [hunk 1]
```diff
@@ -565,7 +565,15 @@ fn layout_atomic<'box_tree>(
...
 
-    ifc.inline_position += pbm.inline_end;
+    ifc.inline_position += pbm.inline_end + fragment.content_rect.size.inline;
+    ifc.current_nesting_level
+        .max_block_size_of_fragments_so_far
+        .max_assign(
+            fragment.content_rect.size.block +
+                fragment.padding.block_sum() +
+                fragment.border.block_sum() +
+                fragment.margin.block_sum(),
```
Comment (SimonSapin) [inline components/layout_2020/flow/inline.rs]: Nit: this is also `pbm.block_sum()`

**Review by SimonSapin — APPROVED** [review]
Nice catch! r+ with one change to save a couple additions

**Comment by SimonSapin** [comment]
Sorry, let’s not land this as-is after all since it seems not to impact any test.

@bors-servo r-

In case this was because relevant tests are not enabled I’m trying this PR on top of https://github.com/servo/servo/pull/25849. It’s still running but looking good so far.

**Comment by SimonSapin** [comment]
Look at that ✨

```
Ran 9459 tests finished in 598.0 seconds.
  • 8860 ran as expected. 0 tests skipped.
  • 15 tests failed unexpectedly
  • 583 tests passed unexpectedly
  • 1 tests had unexpected subtest results

**Comment by SimonSapin** [comment]
Oh. An even less surprising explanation: the previous try run was testing Layout 2013

@bors-servo try=wpt-2020

**Comment by SimonSapin** [comment]
Without https://github.com/servo/servo/pull/25849 this time, still awsome!

```
  • 3312 ran as expected. 0 tests skipped.
  • 1 tests had errors unexpectedly
  • 3 tests failed unexpectedly
  • 154 tests passed unexpectedly
  • 1 tests unexpectedly okay
  • 4 tests had unexpected subtest results
```

You should be able to find the `test-wpt.log` artifact from CI, click "Raw log" in the bottom right menu to find the URL https://community.taskcluster-artifacts.net/QhRdhd11RaedIfm_-oFY0g/0/public/test-wpt.log, then run `./mach update-wpt --layout-2020` with that file.

**Comment by SimonSapin** [comment]
r+ with updated test expectations

**Comment by ferjm** [comment]
> Oh. An even less surprising explanation: the previous try run was testing Layout 2013
> 
> @bors-servo try=wpt-2020

Aha! TIL

**Comment by jdm** [comment]
```
  ▶ PASS [expected FAIL] /css/CSS2/generated-content/before-content-display-005.xht
```

**Comment by ferjm** [comment]
@bors-retry
-  #24786

**Comment by jdm** [comment]
```
  ▶ PASS [expected FAIL] /css/CSS2/generated-content/content-177.xht
```
:/

**Comment by ferjm** [comment]
@bors-servo retry
- #25476

**Pull request merged**